crux_kv/src/lib.rs

```diff
@@ -51,33 +67,34 @@ where
     }
 
     /// Read a value under `key`, will dispatch the event with a
-    /// `KeyValueOutput::Read(Option<Vec<u8>>)` as payload
-    pub fn read<F>(&self, key: &str, make_event: F)
+    /// `KeyValueOutput::Get(KeyValueResult<Option<Vec<u8>>>)` as payload
+    pub fn get<F>(&self, key: &str, make_event: F)
     where
         F: Fn(KeyValueOutput) -> Ev + Send + Sync + 'static,
     {
         let ctx = self.context.clone();
         let key = key.to_string();
         self.context.spawn(async move {
-            let output = ctx.request_from_shell(KeyValueOperation::Read(key)).await;
+            let output = ctx.request_from_shell(KeyValueOperation::Get { key }).await;
 
             ctx.update_app(make_event(output))
         });
     }
 
     /// Read a value under `key`, while in an async context. This is used together with
     /// [`crux_core::compose::Compose`].
-    pub async fn read_async(&self, key: &str) -> KeyValueOutput {
+    pub async fn get_async(&self, key: &str) -> KeyValueOutput {
+        let key = key.to_string();
         self.context
-            .request_from_shell(KeyValueOperation::Read(key.to_string()))
+            .request_from_shell(KeyValueOperation::Get { key })
             .await
     }
 
     /// Set `key` to be the provided `value`. Typically the bytes would be
     /// a value serialized/deserialized by the app.
     ///
-    /// Will dispatch the event with a `KeyValueOutput::Write(bool)` as payload
-    pub fn write<F>(&self, key: &str, value: Vec<u8>, make_event: F)
+    /// Will dispatch the event with a `KeyValueOutput::Set { result: KeyValueResult<()> }` as payload
+    pub fn set<F>(&self, key: &str, value: Vec<u8>, make_event: F)
```

Comment:
Can this be written in terms of `set_async` or am I missing some important detail?

Reply:
Same for the other sync APIs

Reply:
yep, I did that, and it's much better. Also some assertions that the response types are correct